### ISSUE TITLE
Mix of versions required in README.md and versions.tf (and releases info)

### ISSUE THREAD

**Issue by froblesmartin** [terraform-google-modules/terraform-google-kubernetes-engine#1067]
### TL;DR

The versions in README.md files and the required ones in versions.tf do not match. For example in the beta part:
- https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/beta-public-cluster/README.md?plain=1#L317
- https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/beta-public-cluster/versions.tf#L24

And the regular part:
- https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/README.md?plain=1#L274
- https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/versions.tf#L24

Not only the Terraform Google provider but also the Terraform version seems to be wrong as only >= 0.13 is supported as of the rest of the docs.

### Expected behavior

Have consistency in the versions.

### Observed behavior

n/a

### Terraform Configuration

```hcl
n/a
```

### Terraform Version

```sh
n/a
```

### Additional information

I would have opened an MR fixing some of that but I saw this PR which should move the project to use the Terraform Google provider v4: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1066
